### DISCUSSION TITLE
HL7 parser broken in Filter editor

### DISCUSSION THREAD

**Discussion by sfmans** [nextgenhealthcare/connect#5094] [General]
Mirth 3.12.0

Previous versions allowed pasting an HL7 message into the Message Templates area to produce a map of the HL7 elements. 

This functionality is broken in 3.12.0; the bottom status indicator shows activity and 'Parsing...' but the parse does not complete, and a map is not shown in the 'Message Trees' tab

**Comment by pacmano1** [comment]
@sfmans I tried this and can't replicate it.   Maybe post the HL7 message you had an issue with if possible?   I was pasting an HL7 2.4 message and 2.5.1 with no issues.

**Comment by sfmans** [comment]
@pacmano1 thanks. 

MSH|^~\&|PatientCentre|PAS|Graphnet|Graphnet|20211031064512||ADT^A13|20211031064512DS04392||2.4|||NE|AL
EVN|A13|20211031064512||142
PID|||ZZ34448^^^^CN~001129691^^^^UID~01134448^^^^HOSP||PATIENTRACK^PERCY^^MR^^^||19850401|M|||CHEADLE VILLAGE DENT^89B HIGH STREET^CHEADLE^CHESHIRE^SK8 1AA^^PERM^DHA^QOP||01612223333^PH^HOME~^INTERNET^NET|01614445555^PH^WP||S|C^|^^^^||||A|Stockport||||Test Occupation|||0|07
PD1|||GATELY MEDICAL CEN^^P88024^|G9104927^DAVISON^A^^^DR^^^OLD HALL ROAD^GATLEY^CHEADLE^SK8 4DG
NK1|1|Penny Patientrack|SISTER|CRUCIAL INTERIORS^87 HIGH STREET^CHEADLE^CHESHIRE^SK8 1AA|01615558888^PH^HOME|01617779999^PH^WP
PV1||I|D1^^04^^^^^^D1 - GREEN - DAY CASE||||BOD^DARMAS^B^^^MR||BOD^DARMAS^B^^^MR|SURG^GENERAL SURGERY||||HO^USUAL ADDRESS/NFA|||BOD^DARMAS^B^^^MR||011344483|||||||||||||||||NO^NORMAL|HO^USUAL ADDRESS/NFA||||||||202108201310

(Screenshot shortened but no detail other than whitespace omitted)
![image](https://user-images.githubusercontent.com/83290451/140276722-9a7b31f9-b3dd-463e-95a7-1fdd499c786b.png)

**Comment by pacmano1** [comment]
That works fine here also.

Message you pasted:

```text
MSH|^&|PatientCentre|PAS|Graphnet|Graphnet|20211031064512||ADT^A13|20211031064512DS04392||2.4|||NE|AL
EVN|A13|20211031064512||142
PID|||ZZ34448^^^^CN001129691^^^^UID01134448^^^^HOSP||PATIENTRACK^PERCY^^MR^^^||19850401|M|||CHEADLE VILLAGE DENT^89B HIGH STREET^CHEADLE^CHESHIRE^SK8 1AA^^PERM^DHA^QOP||01612223333^PH^HOME^INTERNET^NET|01614445555^PH^WP||S|C^|^^^^||||A|Stockport||||Test Occupation|||0|07
PD1|||GATELY MEDICAL CEN^^P88024^|G9104927^DAVISON^A^^^DR^^^OLD HALL ROAD^GATLEY^CHEADLE^SK8 4DG
NK1|1|Penny Patientrack|SISTER|CRUCIAL INTERIORS^87 HIGH STREET^CHEADLE^CHESHIRE^SK8 1AA|01615558888^PH^HOME|01617779999^PH^WP
PV1||I|D1^^04^^^^^^D1 - GREEN - DAY CASE||||BOD^DARMAS^B^^^MR||BOD^DARMAS^B^^^MR|SURG^GENERAL SURGERY||||HO^USUAL ADDRESS/NFA|||BOD^DARMAS^B^^^MR||011344483|||||||||||||||||NO^NORMAL|HO^USUAL ADDRESS/NFA||||||||202108201310
```
And parsed:

![image](https://user-images.githubusercontent.com/44065187/140305571-25f4fddb-fb1d-4898-8cde-ad8fc37ed036.png)

That works in both a filter and transformer.

**Comment by pacmano1** [comment]
Just a note, see https://github.com/nextgenhealthcare/connect/discussions/4824 and the actual instrucs for formatting code: https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.
